search/search-client-query-transformer/src/main/java/io/camunda/search/clients/query/SearchQueryBuilders.java

```diff
@@ -21,6 +24,10 @@
 
 public final class SearchQueryBuilders {
 
+  public static final String DATE_TIME_FORMAT = "yyyy-MM-dd'T'HH:mm:ss.SSSZZ";
+  private static final DateTimeFormatter DATE_TIME_FORMATTER =
+      DateTimeFormatter.ofPattern(DATE_TIME_FORMAT);
+
```

Comment:
> ❓ We are preparing the OffsetDateTime to a string that the respective database can understand, right? Should this conversion happen in the database-specific converters if this differs per database? What do you think?

That could make sense.
In the change I replicated what the [`DateValueFilterTransformer`](https://github.com/camunda/camunda/blob/main/search/search-client-query-transformer/src/main/java/io/camunda/search/clients/transformers/filter/DateValueFilterTransformer.java) did previously but for `Operation`.
Maybe this could be a follow-up?

I will replace it with `ISO_OFFSET_DATE_TIME` though, that's much better.

Reply:
I see. Let's keep it in the general transformer then for now. If this works for all supported databases, fine. Otherwise, we could follow up later and revise this for databases that cannot handle this (but that should indeed be very unlikely since it's a standard).

Reply:
`ISO_OFFSET_DATE_TIME` also behaves slightly different. It can omit seconds, nanos if they are zero:

```
expected: "2024-03-12T10:30:15.000+0000"
 but was: "2024-03-12T10:30:15Z"
```

Reply:
Oh, yeah, I forgot about that. That pattern uses optional fractions. That might however be totally fine for the databases and the resulting queries, though 😅